tests/store/model_registry/test_sqlalchemy_store.py

```diff
@@ -947,6 +936,54 @@ def search_versions(filter_string):
         assert mvds[0].source == "A/B"
         assert mvds[0].description == "Online prediction model!"
 
+    def test_search_model_versions_in_clause_mysql_case_sensitive(self):
+        from mlflow.utils.search_utils import SearchModelUtils
+        from mlflow.store.db.db_types import MYSQL
+        run_id1 = uuid.uuid4().hex
+        run_id2 = uuid.uuid4().hex
+        filter_str = f"run_id IN ('{run_id1.upper()}','{run_id2}')"
+        parsed_filter = SearchModelUtils.parse_search_filter(filter_str)
+        attr_filters, _ = SqlAlchemyStore._get_search_model_versions_filter_clauses(parsed_filter, MYSQL)
+        self.assertEqual(
+            attr_filters[0].right.effective_value,
+            [run_id2]
+        )
+
+    def test_search_model_versions_by_tag(self):
+        # create some model versions
+        name = "test_for_search_MV_by_tag"
+        self._rm_maker(name)
+        run_id_1 = uuid.uuid4().hex
+        run_id_2 = uuid.uuid4().hex
+
+        mv1 = self._mv_maker(
+            name=name, source="A/B", run_id=run_id_1,
+            tags=[ModelVersionTag("t1", "abc"), ModelVersionTag("t2", "xyz")]
+        )
+        self.assertEqual(mv1.version, 1)
+        mv2 = self._mv_maker(
+            name=name, source="A/C", run_id=run_id_2,
+            tags=[ModelVersionTag("t1", "abc"), ModelVersionTag("t2", "x123")]
+        )
+        self.assertEqual(mv2.version, 2)
+
+        def search_versions(filter_string):
+            return [mvd.version for mvd in self.store.search_model_versions(filter_string)]
+
+        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 = 'xyz'")), {1})
+        self.assertEqual(set(search_versions(f"name = 'wrong_name' and tag.t2 = 'xyz'")), set())
+        self.assertEqual(set(search_versions(f"name = '{name}' and tag.`t2` = 'xyz'")), {1})
+        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t3 = 'xyz'")), set())
+        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 != 'xy'")), {1, 2})
+        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 LIKE 'xy%'")), {1})
+        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 LIKE 'xY%'")), set())
+        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 ILIKE 'xY%'")), {1})
+        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 LIKE 'x%'")), {1, 2})
+        self.assertEqual(set(search_versions(f"name = '{name}' and tag.T2 = 'xyz'")), set())
```

Comment:
Why this test passed ?

Reply:
See https://github.com/mlflow/mlflow/runs/7535341371?check_suite_focus=true. `tests/store/model_registry/test_sqlalchemy_store.py` is not tested on MySQL. You need to merge master.

Reply:
The similar test in databricks backend is also passed:
https://github.com/databricks/universe/pull/193846/files#r936134623

Reply:
Sounds like this isn't an issue

Reply:
It turns out this `tests/store/model_registry/test_sqlalchemy_store.py` is not executed on MySQL. I manually modified the code, ran this line on MySQL and got this error:

```
______________________________________________________________________________________________________________________________________ TestSqlAlchemyStoreSqlite.test_search_model_versions_by_tag _______________________________________________________________________________________________________________________________________

self = <test_sqlalchemy_store.TestSqlAlchemyStoreSqlite testMethod=test_search_model_versions_by_tag>

    def test_search_model_versions_by_tag(self):
        # create some model versions
        name = "test_for_search_MV_by_tag"
        self._rm_maker(name)
        run_id_1 = uuid.uuid4().hex
        run_id_2 = uuid.uuid4().hex
    
        mv1 = self._mv_maker(
            name=name,
            source="A/B",
            run_id=run_id_1,
            tags=[ModelVersionTag("t1", "abc"), ModelVersionTag("t2", "xyz")],
        )
        self.assertEqual(mv1.version, 1)
        mv2 = self._mv_maker(
            name=name,
            source="A/C",
            run_id=run_id_2,
            tags=[ModelVersionTag("t1", "abc"), ModelVersionTag("t2", "x123")],
        )
        self.assertEqual(mv2.version, 2)
    
        def search_versions(filter_string):
            return [mvd.version for mvd in self.store.search_model_versions(filter_string)]
    
        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 = 'xyz'")), {1})
        self.assertEqual(set(search_versions(f"name = 'wrong_name' and tag.t2 = 'xyz'")), set())
        self.assertEqual(set(search_versions(f"name = '{name}' and tag.`t2` = 'xyz'")), {1})
        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t3 = 'xyz'")), set())
        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 != 'xy'")), {1, 2})
        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 LIKE 'xy%'")), {1})
        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 LIKE 'xY%'")), set())
        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 ILIKE 'xY%'")), {1})
        self.assertEqual(set(search_versions(f"name = '{name}' and tag.t2 LIKE 'x%'")), {1, 2})
>       self.assertEqual(set(search_versions(f"name = '{name}' and tag.T2 = 'xyz'")), set())
E       AssertionError: Items in the first set but not the second:
E       1
```

Reply:
@dbczumar @WeichenXu123 I'm looking at `case_sensitive_mysql_eq`:

https://github.com/mlflow/mlflow/blob/aa9ad78d175839144d40237ef662a9f8a0e9e4e6/mlflow/utils/search_utils.py#L109

```
        def case_sensitive_mysql_eq(value):
            return sa.and_(column.__eq__(value), column.op("= BINARY", is_comparison=True)(value))
```

This is compiled to `<column> = 'foo' AND <column> = BINARY 'foo'`, but what we want is `<column> = 'foo' AND BINARY <column> = 'foo'`. I'll file a PR.